cachedb/redis.c

```diff
@@ -114,6 +118,24 @@ redis_init(struct module_env* env, struct cachedb_env* cachedb_env)
 	for(i = 0; i < moddata->numctxs; i++)
 		moddata->ctxs[i] = redis_connect(moddata);
 	cachedb_env->backend_data = moddata;
+
+	if (env->cfg->redis_set_ttl) {
+		/** check if setex command is supported */
+		rep = redis_command(env, cachedb_env, "SETEX __UNBOUND_REDIS_CHECK__ 1 none", NULL, 0);
+		if(!rep)
+			/** init failed, no response from redis server*/
+			log_err("redis_init: failed to init redis, could not connect to server");
+			return 0;
+		switch (rep->type) {
```

Comment:
From the [documentation](https://redis.io/commands/setex) SETEX returns only a `REDIS_REPLY_STRING`. I would break on that (what you currently do with default) and error out on default instead.

Reply:
I was not able to apply this change. I tested it but somehow redis (keydb in my case, which is 100% compatible with redis) returns "5" as response type, which is not defined in hiredis library. That is why I kept it the way it was before. Maybe you have an idea why it behaves that way.

Reply:
My bad, 5 (REDIS_REPLY_STATUS) is the correct value. The documentation was mentioning "simple string reply" and I hastily thought REDIS_REPLY_STRING was the one.

Reply:
Very good hint! I have checked the "hiredis" code but was not able to find the constant with value "5". I will add the change. Thanks!

Reply:
I found it [here](https://github.com/redis/hiredis/blob/master/read.h#L55)

Reply:
Ok, again missed something "obvious"... sorry!